platform/posix/utest/reconnect_utest.c

```diff
@@ -0,0 +1,172 @@
+#include <string.h>
+#include <stdbool.h>
+#include <stdlib.h>
+#include "/usr/include/errno.h"
+
+#include "unity.h"
+
+/* Include paths for public enums, structures, and macros. */
+#include "transport_reconnect.h"
+
+#include "mock_time_api.h"
+#include "mock_math_api.h"
+
+/* Return value of mocked #rand. */
+#define RAND_RET_VAL            ( MAX_JITTER_VALUE_SECONDS + ( MAX_JITTER_VALUE_SECONDS / ( MAX_RECONNECT_ATTEMPTS ) ) )
+#define EXPECTED_NEXT_JITTER    ( RAND_RET_VAL % MAX_JITTER_VALUE_SECONDS )
+/* Parameters to track the next max jitter or number of attempts done. */
+static TransportReconnectParams_t reconnectParams;
+/* Return value of #Transport_ReconnectBackoffAndSleep. */
+static bool retriesArePending;
+
+/* ============================   UNITY FIXTURES ============================ */
+
+/* Called before each test method. */
+void setUp()
+{
+}
+
+/* Called after each test method. */
+void tearDown()
+{
+    retriesArePending = false;
+}
+
+/* Called at the beginning of the whole suite. */
+void suiteSetUp()
+{
+}
+
+/* Called at the end of the whole suite. */
+int suiteTearDown( int numFailures )
+{
+    return numFailures;
+}
+
+/* ========================================================================== */
+
+/**
+ * @brief A helper method to make assertions on the #TransportReconnectParams_t
+ * object upon calling #Transport_ReconnectParamsReset.
+ */
+static void verifyReconnectParamsAfterReset( void )
+{
+    TEST_ASSERT_EQUAL( 0, reconnectParams.attemptsDone );
+    TEST_ASSERT_EQUAL( EXPECTED_NEXT_JITTER, reconnectParams.nextJitterMax - EXPECTED_NEXT_JITTER );
```

Comment:
Suggestion to improve readability of this assert
```suggestion
    TEST_ASSERT_EQUAL( INITIAL_RECONNECT_BACKOFF_SECONDS + EXPECTED_NEXT_JITTER , reconnectParams.nextJitterMax );
```